core/src/main/java/org/elasticsearch/common/lucene/uid/PerThreadIDVersionAndSeqNoLookup.java

```diff
@@ -65,9 +71,24 @@
     PerThreadIDVersionAndSeqNoLookup(LeafReader reader) throws IOException {
```

Comment:
I didn't look very close here but can we keep to to one enum somehow and pass in the field mapper name as a ctor argument? it should stay the same across the lifetime of an index?